### PR TITLE
fix(VDataTable): union column filter with search results

### LINKED ISSUE #11402: [Bug Report] VDatatable: Custom filtering not working
### Environment
**Vuetify Version:** 2.2.28
**Last working version:** 2.2.26
**Vue Version:** 2.6.11
**Browsers:** Firefox 75.0, Google Chrome
**OS:** Mac OS 10.14

### Steps to reproduce
1. Navigate to Custom filter example in Vuetify website: 
https://vuetifyjs.com/en/components/data-tables/#custom-filtering

2. Entering data in search box is not triggering filtering functionality.

### Expected Behavior
Entering data in search box should filter datatable results.

### Actual Behavior
Nothing happens.

### Reproduction Link
<a href="https://codepen.io/rw-leo/pen/OJyweQp?editable=true&editors=101%3Dhttps%3A%2F%2Fvuetifyjs.com%2Fen%2Fcomponents%2Fdata-tables%2F" target="_blank">https://codepen.io/rw-leo/pen/OJyweQp?editable=true&editors=101%3Dhttps%3A%2F%2Fvuetifyjs.com%2Fen%2Fcomponents%2Fdata-tables%2F</a>

### Other comments
search functionality seems to be working in datatables, as long as not being used along with custom-filter.

### PR DESCRIPTION
**Pull request by mk86**
## Description
Ensure that filters and search terms are combined to display the intersection of results, rather than the union - so each filter applied makes the set of matches smaller for refined searching.

fixes #11402 

## Motivation and Context
There is some discussion on issue #11402 , but the current behaviour is very surprising and unlikely to be what a majority of users would want.
Note that I have not marked this as a breaking change, though it changes behaviour - as in the wording below "would cause existing functionality to not work as expected", I don't believe the current behaviour is what would be expected..

## How Has This Been Tested?
Unit tests included for the case where we have column filtering only, search term only and both. I replaced a previous test which would no longer pass, since it would include items which matched EITHER search term OR column filter.
Note that I added two of the tests at the end as the snapshot IDs changed otherwise and I thought it best not to require regenerating that as part of this.

## Markup:
(See unit tests)

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library) (no changes, since they are already inline with the change)

### CODE REVIEW

**Comment by mk86** [comment]
Very sorry about that, it seems my local settings didn't pick that up. Not sure why it's not taking settings from the repo - will take a look into it! Fix pushed for now.

**Comment by mk86** [comment]
Travis is showing the build as passing - not sure why the checks status isn't updating. Can we kick it or do I need to push something else to wake it up?

**Comment by mk86** [comment]
Merging from master appears to have kicked it into action! :)

File: packages/vuetify/src/components/VDataTable/VDataTable.ts [hunk 1]
```diff
@@ -51,17 +51,16 @@ function searchTableItems (
...
-  // If the `search` property is empty and there are no custom filters in use, there is nothing to do.
-  if (!(search && headersWithoutCustomFilters.length) && !headersWithCustomFilters.length) return items
-
   return items.filter(item => {
     // Headers with custom filters are evaluated whether or not a search term has been provided.
-    if (headersWithCustomFilters.length && headersWithCustomFilters.every(filterFn(item, search, defaultFilter))) {
-      return true
-    }
+    // We need to match every filter to be included in the results.
+    const matchesColumnFilters = headersWithCustomFilters.every(filterFn(item, search, defaultFilter))
```
Comment (deepapanwar) [inline packages/vuetify/src/components/VDataTable/VDataTable.ts:57]: customFilter should be applied here.

File: packages/vuetify/src/components/VDataTable/VDataTable.ts [hunk 2]
```diff
@@ -51,17 +51,16 @@ function searchTableItems (
...
+
+    // Headers without custom filters are only filtered by the `search` property if it is defined.
+    // We only need a single column to match the search term to be included in the results.
+    const matchesSearchTerm = !search || headersWithoutCustomFilters.some(filterFn(item, search, customFilter))
```
Comment (deepapanwar) [inline packages/vuetify/src/components/VDataTable/VDataTable.ts:61]: defaultFilter should be applied here

**Pull request merged**